external-crates/move/crates/move-compiler/src/diagnostics/mod.rs

```diff
@@ -103,11 +104,56 @@ enum MigrationChange {
 
 // All of the migration changes
 pub struct Migration {
-    files: SimpleFiles<Symbol, String>,
-    file_mapping: FileMapping,
+    mapped_files: MappedFiles,
     changes: BTreeMap<FileId, BTreeMap<usize, Vec<(usize, MigrationChange)>>>,
 }
 
+/// A mapping from file ids to file contents along with the mapping of filehash to fileID.
+pub struct MappedFiles {
+    files: SimpleFiles<Symbol, Arc<str>>,
+    file_mapping: FileMapping,
+}
+
+impl MappedFiles {
+    pub fn new(files: FilesSourceText) -> Self {
+        let mut simple_files = SimpleFiles::new();
+        let mut file_mapping = HashMap::new();
+        for (fhash, (fname, source)) in files {
+            let id = simple_files.add(fname, source);
+            file_mapping.insert(fhash, id);
+        }
+        Self {
+            files: simple_files,
+            file_mapping,
+        }
+    }
```

Comment:
These file changes are mostly what I was referring to as a separate diff

Reply:
Split off into their own PR here: https://github.com/MystenLabs/sui/pull/16758